Infrastructure/LookupDataService.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Application;
```

Comment:
Do we need this service? I suggest we create a Mediator query, i.e. `GetSubjectsAsyncQuery` with a handler to get the data and inject the `ILookupDataRepository` to fetch the data from the repository. Then we can use `await _mediator.Send(query);` to get the data we need.

Reply:
This has been done to be consistent with the other services (e.g. TuitionPartnerService and LookupDataService).  I think that removing the services and going straight to repositories is something we should consider, but is a wider refactor job to ensure it is consistent.  
I think that the calls to the service are already done async via mediator, but to the service.
I did notice that this file was not in the same folder as the TuitionPartnerService and LookupDataService classes, so have moved it there.

Reply:
Okay, I suggested it because we are adding a new service. We may have to refactor these at some point (i.e., TuitionPartnerService and LookupDataService) and use the Mediator query or command. I am not suggesting refactoring the existing other services but not adding new ones.

Reply:
That lookup repository wasn't being called prior to this, but was in existence.  So to use it I either 

a) stick with the existing pattern of going via a service and accept that I'm adding it, even if it might be removed later
b) just call the repository directly and don't refactor which will be a different pattern to the existing
c) refactor all to use the approach we want to use.  

I'd prefer to stick with a) for now and move to c) when we have a chance.